File: data-jdbc/src/test/groovy/io/micronaut/data/jdbc/h2/remap/Course.java
package io.micronaut.data.jdbc.h2.remap;

import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.MappedEntity;
import io.micronaut.data.annotation.Relation;

import java.util.List;
import java.util.UUID;

import static io.micronaut.data.annotation.Relation.Cascade.ALL;
import static io.micronaut.data.annotation.Relation.Kind.MANY_TO_MANY;

@MappedEntity("att_course")
record Course(

    @Id
    UUID id,

    String name,

    @Relation(value = MANY_TO_MANY, mappedBy = "courses", cascade = ALL)
    List<Student> students
) {
}

File: doc-examples/mongo-reactive-example-kotlin/src/main/kotlin/example/Course.kt
package example

import io.micronaut.data.annotation.GeneratedValue
import io.micronaut.data.annotation.Id
import io.micronaut.data.annotation.MappedEntity
import io.micronaut.data.annotation.Relation
import org.bson.types.ObjectId

@MappedEntity
data class Course(
        @field:Id @GeneratedValue
        val id: ObjectId?,
        val name: String,
        @Relation(value = Relation.Kind.MANY_TO_MANY, mappedBy = "courses")
        val students: List<Student>
) {
    constructor(name: String) : this(null, name, emptyList())
}
